src/Features/Core/Portable/ConvertAnonymousTypeToClass/AbstractConvertAnonymousTypeToClassCodeRefactoringProvider.cs

```diff
@@ -70,34 +70,13 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
         private async Task<(TAnonymousObjectCreationExpressionSyntax, INamedTypeSymbol)> TryGetAnonymousObjectAsync(
             Document document, TextSpan span, CancellationToken cancellationToken)
         {
-            var position = span.Start;
-            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var token = root.FindToken(position);
-
-            // Span actually has to be within the token (i.e. not in trivia around it).
-            if (!token.Span.IntersectsWith(position))
-            {
-                return default;
-            }
-
-            if (!span.IsEmpty && span != token.Span)
-            {
-                // if there is a selection, it has to be of the whole token.
-                return default;
-            }
-
-            var anonymousObject = token.Parent as TAnonymousObjectCreationExpressionSyntax;
+            var refactoringHelperService = document.GetLanguageService<IRefactoringHelpersService>();
+            var anonymousObject = await refactoringHelperService.TryGetSelectedNodeAsync<TAnonymousObjectCreationExpressionSyntax>(document, span, cancellationToken).ConfigureAwait(false);
```

Comment:
was there an updated test for this?  i.e. selecting an entire object creation expr?

Reply:
Nope, will add.